### PR TITLE
Mentor comments

### PR DESCRIPTION
**Pull request by VladSelikh**

### CODE REVIEW

File: test/specs/spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const expect = require("chai").expect;
+const PageFactory = require("../utils/pageFactory");
+const EC = protractor.ExpectedConditions;
+describe("Heroes home page tests", function () {
+
+    beforeEach(function() {
+            browser.ignoreSynchronization = true;
+            return browser.manage().window().maximize();
```
Comment (VladSelikh) [inline test/specs/spec.js]: You don't need `return` statement here.

File: test/specs/spec.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    });
+
+    it("should have 11 recent badges", async function () {
+        await PageFactory.getPage("Login").open();
```
Comment (VladSelikh) [inline test/specs/spec.js]: Please move pages declaration to separate constants:
```
const loginPage = Pagefactory.getPage("Login");
// some code
await loginPage.open();
await logiPAge.typrLogin();

File: test/utils/base_elements/base_collection.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        return arrayOfCollectionTexts;
+    };
+    async clickElementByText(textToClick) {
+        const arrayOfElementTexts = await this.collection.getText();
+        const elementToClickIndex = arrayOfElementTexts.indexOf(textToClick);
+        if (elementToClickIndex === -1) {
+            throw new Error(`No element with [${textToClick}] text found!`);
+        }
+        logger.info(`Clicking "${textToClick}" text in "${this.elementName}"`);
+        return this.collection.get(elementToClickIndex).click();
```
Comment (VladSelikh) [inline test/utils/base_elements/base_collection.js]: There is no need to return this statement.

File: test/utils/base_page/base_page.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const logger = require('../../../test/config/logger.config.js');
+const Header = require('./header');
+
+class BasePage {
+    constructor() {
+        this.Header = new Header();
+    };
+    wait(seconds) {
```
Comment (VladSelikh) [inline test/utils/base_page/base_page.js]: Please don't break the Single responsibility principle here: move the `wait()` method to a separate function helper.

File: test/utils/base_page/base_page.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        logger.info(`Waiting "${seconds * 1000}" milliseconds`);
+        return browser.sleep(seconds * 1000);
+    };
+    async getCurrenUrl() {
```
Comment (VladSelikh) [inline test/utils/base_page/base_page.js]: Please move `getCurrentUrl()` method to a separate function helper.

File: test/utils/base_page/base_page.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        const currentUrl = browser.getCurrentUrl();
+        logger.debug(`Current url is "${currentUrl}"`);
+        return currentUrl;
+    };
+    open(url) {
```
Comment (VladSelikh) [inline test/utils/base_page/base_page.js]: In this case it is better to do on e of these 2 options:
1) Add `this.url` to your constructor which will be your page URL and than use it in this method:
```
open() {
   logger.info(`Opening "${this.url}" url`);
   browser.get(this.url);
}
```
2) Move the whole method to a separate function helpter:
```
async function openPage(url) {
   logger.info(`Opening "${url}" url`);
   browser.get(url);
}

File: test/utils/login_page/credentials.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+const login = "my_login";
+const password = "my_password";
+
+module.exports = {login, password};
```
Comment (VladSelikh) [inline test/utils/login_page/credentials.js]: Please move `credentials.js` folder to a separate `constants` folder under `test/utils/constants` path.

**Comment by VladSelikh** [comment]
Common comments:
1) Please don't use `snake` notation while naming your page objects. As we work with JavaScript we use `lowerCamelCase` notation while naming files and folders.
2) Please delete all `return` keywords when the code to the right of `return` returns `void`.
3) You use only css locators when identifying elements on the page. Use xpath also (id, className if you like). A little hint:
you can rewrite your constructor to accept selector type as a third argument:
```
constructor(elementName, selectorType, selector) {
        // some code
    }

File: test/specs/spec.js (hunk 1 above)
Comment (aliaksandr-harunou) [reply test/specs/spec.js]: fixed

File: test/specs/spec.js (hunk 2 above)
Comment (aliaksandr-harunou) [reply test/specs/spec.js]: done

File: test/utils/base_page/base_page.js (hunk 4 above)
Comment (aliaksandr-harunou) [reply test/utils/base_page/base_page.js]: done

File: test/utils/base_page/base_page.js (hunk 6 above)
Comment (aliaksandr-harunou) [reply test/utils/base_page/base_page.js]: done

File: test/utils/base_elements/base_collection.js (hunk 3 above)
Comment (aliaksandr-harunou) [reply test/utils/base_elements/base_collection.js]: fixed

File: test/utils/base_page/base_page.js (hunk 5 above)
Comment (aliaksandr-harunou) [reply test/utils/base_page/base_page.js]: done

File: test/utils/login_page/credentials.js (hunk 7 above)
Comment (aliaksandr-harunou) [reply test/utils/login_page/credentials.js]: done

**Pull request merged**